android/build.gradle

```diff
@@ -9,14 +9,18 @@ buildscript {
     }
 }
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 26
+    compileSdkVersion safeExtGet('compileSdkVersion', 26)
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 26
+        minSdkVersion safeExtGet('minSdkVersion', 16)
```

Comment:
I don't think we want to inherit this value from the parent project; this would allow this library to be used by API versions below 16, which we don't currently support. I certainly see the value in inheriting the other two SDK version-related parameters, though!

Reply:
That makes sense. What do you think about something like `safeExtGetWithMin('minSdkVersion', 16)` which does something like this:
```
if(!propExistsInParent) 
   return defaultAndMinValue
if(propInParent >= defaultAndMinValue) 
   return propInParent
return defaultAndMinValue
```

Reply:
I'm struggling to think of a reason `minSdkVersion` would need to be parameterized at all:

- If the parent project has `minSdkVersion` < 16, this library should (rightly) refuse to compile.
- If the parent project has `minSdkVersion` == 16, there are no issues.
- If the parent project has `minSdkVersion` > 16, that doesn't cause any issues either since this only specifies the minimum version.

I'm happy to let this conditional in if you have a concrete use case for it, but I'd prefer to avoid the complexity unless absolutely necessary.

Reply:
I think you're right @timothyandrew, having that seems unnecessary when you think about it. I'll remove that.

Reply:
Sounds good, thank you! 👍